### PR TITLE
Seed course data on server start up

### PR DESCRIPTION
**Pull request by evenskari**
This includes
- textfile scraping data from the ucsb website
- seed.db parsing said data
- configuration of elastic beanstalk

### CODE REVIEW

**Comment by dannycho7** [comment]
I moved the data.txt file to the db folder and called it schools_data.json. I think it's good to make sure that'll still work with the seeds.rb file.

I also removed the extra columns from our models. It adds some unnecessary complexity, but we can add them in the future if we find a need for them.

**Comment by dannycho7** [comment]
I would recommend adding the is_graduate_course column to the course table.

File: db/seeds.rb [hunk 1]
```diff
@@ -1,7 +1,45 @@
...
+  units= parsed_course["units"]
+  description= parsed_course["desc"]
+  instructor_name= parsed_course["instructors"] # returns instructors separated by comma
+  is_graduate_course = parsed_course['is_grad']
+  # Unused data: students_enrolled_actually, prequisites
+  if instructor_name.include? ","
+    instructor_name = instructor_name.split(",")[0]
+  end
+  instructor = Instructor.where(id: instructor_name).first_or_create(rmp_url: "lol")
+  course = instructor.courses.where(dept: department, course_no: id).first_or_create(description: description,
```
Comment (estensen) [inline db/seeds.rb]: Reformat so lines are 80 characters or less. It's hard to read the code when you split your editor to view multiple files

File: db/seeds.rb [hunk 2]
```diff
@@ -1,7 +1,45 @@
...
-# Examples:
-#
-#   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
-#   Character.create(name: 'Luke', movie: movies.first)
+require 'json'
+File.open("db/schools_data.json").each do |line|
+  parsed_course = JSON.parse(line) #Every line in a the .txt file represents an JSON course object with belonging time objects
+  department = parsed_course["department"]
+  grading_option = parsed_course["grading_option"]
+  # name = parsed_course["name"]
```
Comment (estensen) [inline db/seeds.rb]: Why isn't the course name part of the schema? We need that for the table we're displaying.

Comment (evenskari) [reply db/seeds.rb]: Yeah, it was removed due to a request from Danny. Will add it back

File: db/seeds.rb [hunk 3]
```diff
@@ -1,7 +1,54 @@
...
+  is_graduate_course = parsed_course['is_grad']
+  # Unused data: students_enrolled_actually, prerequisites
+  if instructor_name.include? ","
+    instructor_name = instructor_name.split(",")[0]
+  end
+  instructor = Instructor.where(id: instructor_name)
+                         .first_or_create(rmp_url: 'example.com')
+  course = instructor.courses.where(dept: department, course_no: id)
+                     .first_or_create(
+                         name: name,
```
Comment (estensen) [inline db/seeds.rb]: Use consistent formatting. Indent name and description.

File: db/seeds.rb [hunk 4]
```diff
@@ -1,7 +1,54 @@
...
+                         is_graduate_course: is_graduate_course
+                     )
+  parsed_course["timeslots"].each do |time|
+    days = time["days"]
+    room = time["location"]
+    start_time = time["start_time"]
+    end_time = time["end_time"]
+    type = time["type"]
+    # Unused data: enrolled_max, enrolled_count, instructor
+    if start_time == ""
```
Comment (estensen) [inline db/seeds.rb]: It might be better to not define the start_time and end_time. It could be a problem for the calendar.

File: db/seeds.rb [hunk 5]
```diff
@@ -1,7 +1,54 @@
...
+      start_time = "00:00pm"
+    end
+    if end_time == ""
+      end_time = "00:00pm"
+    end
+    course.periods.where(
+        start_time: start_time,
```
Comment (estensen) [inline db/seeds.rb]: Indent for nicer formatting

File: db/seeds.rb [hunk 6]
```diff
@@ -1,7 +1,54 @@
...
-# The data can then be loaded with the rails db:seed command (or created alongside the database with db:setup).
-#
-# Examples:
-#
-#   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
-#   Character.create(name: 'Luke', movie: movies.first)
+require 'json'
+File.open("db/schools_data.json").each do |line|
+  parsed_course = JSON.parse(line)
+  # Every line in a the .txt file represents an JSON course object with
```
Comment (estensen) [inline db/seeds.rb]: Unnecessary comment. It is declared on line 3 that JSON is used

File: db/seeds.rb [hunk 7]
```diff
@@ -1,7 +1,54 @@
...
+  # belonging time objects
+  department = parsed_course["department"]
```
Comment (estensen) [inline db/seeds.rb]: Indent right side for nicer formatting

File: db/seeds.rb [hunk 8]
```diff
@@ -1,7 +1,54 @@
...
+  grading_option = parsed_course["grading_option"]
+  name = parsed_course["name"]
+  id = parsed_course["id"]
+  students_enrolled_max  = parsed_course["max_students"]
+  units= parsed_course["units"]
```
Comment (estensen) [inline db/seeds.rb]: Add space before `=`

File: db/seeds.rb [hunk 9]
```diff
@@ -1,7 +1,54 @@
...
+  description= parsed_course["desc"]
+  instructor_name= parsed_course["instructors"]
+  # returns instructors separated by comma
+  is_graduate_course = parsed_course['is_grad']
+  # Unused data: students_enrolled_actually, prerequisites
+  if instructor_name.include? ","
+    instructor_name = instructor_name.split(",")[0]
+  end
```
Comment (estensen) [inline db/seeds.rb]: Add space after `end`'s. The code is very compact

File: db/seeds.rb (hunk 7 above)
Comment (evenskari) [reply db/seeds.rb]: Do you mean by putting the second part of the comment(line 5) on line 4? Then it breaks 80 characters. If it's about line 6, please explain how.

File: db/seeds.rb (hunk 6 above)
Comment (evenskari) [reply db/seeds.rb]: Removing the comment

File: db/seeds.rb (hunk 8 above)
Comment (evenskari) [reply db/seeds.rb]: Ok

File: db/seeds.rb (hunk 9 above)
Comment (evenskari) [reply db/seeds.rb]: OK

File: db/seeds.rb (hunk 3 above)
Comment (evenskari) [reply db/seeds.rb]: .first_or_create(
                             name: name, 
                             description: description,
                         units:              units,
                         grading_opts:       grading_option,
                         max_class_size:     students_enrolled_max,
                         instructor_id:      instructor_name,
                         is_graduate_course: is_graduate_course
                     )
Like this? Why should the 2 first parameters be indented?

Comment (evenskari) [reply db/seeds.rb]: Click to edit my last comment to see the indentation

File: db/seeds.rb (hunk 4 above)
Comment (evenskari) [reply db/seeds.rb]: As in using those exact variable names?

File: db/seeds.rb (hunk 5 above)
Comment (evenskari) [reply db/seeds.rb]: Added space after end statements, but unsure how the proper formatting  of the methods with multiple parameters should be

File: db/seeds.rb (hunk 7 above)
Comment (evenskari) [reply db/seeds.rb]: Oh, I understand now

**Pull request merged**